### PR TITLE
[FORGE-2317] Improve delete recursive.

### PR DESCRIPTION
**Pull request by danielsoro**

### CODE REVIEW

**Comment by gastaldi** [comment]
This does not take in account when in context of a file transaction, where physical files to not exist yet

**Comment by danielsoro** [comment]
@gastaldi,

This case, we can put the implementation in fileOperator? :)
So you have DefaultFileOperator and FileResourceTransactionImpl.

Have we  test for transaction operation?

**Comment by danielsoro** [comment]
Can I use the same strategy, but in Transaction mode.. I call session.delete 
And the default, work like in PR. :)

WDYT? Sounds good?

**Comment by gastaldi** [comment]
I think moving to DefaultFileOperation is better. About the transaction, I am unsure if session.delete will be enough to delete recursively

**Comment by danielsoro** [comment]
I think to do that:

```
Files.walkFileTree(file.toPath(), new SimpleFileVisitor<Path>()
      {
         @Override
         public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException
         {
            session.delete(file);
            return FileVisitResult.CONTINUE;
         }

         @Override
         public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException
         {
            session.delete(dir);
            return FileVisitResult.CONTINUE;
         }
      });
```

That not work in transactionOperation?
It should to work, not?

**Comment by gastaldi** [comment]
Probably not because the directory that you are referencing in file.toPath() might not exist physically

File: resources/impl/src/main/java/org/jboss/forge/addon/resource/transaction/file/FileResourceTransactionImpl.java [hunk 1]
```diff
@@ -483,6 +467,49 @@ public void deleteFileOnExit(File file)
...
    }
 
+   @Override
+   public boolean deleteFile(File file, boolean recursive)
+   {
+      if (recursive)
+      {
+         return this.deleteRecursive(file);
+      }
+      return file.delete();
```
Comment (gastaldi) [inline resources/impl/src/main/java/org/jboss/forge/addon/resource/transaction/file/FileResourceTransactionImpl.java]: deleteFile(file)

**Comment by gastaldi** [comment]
It looks good now. Why is there a merge conflict?

**Comment by danielsoro** [comment]
Sorry.
I forgot to update my branch. I'll fix the PR.

**Pull request merged**